ut/performance_tests.cpp

```diff
@@ -0,0 +1,233 @@
+              << std::endl;
+
+    timer.Reset();
+    timer.Start();
+    EXPECT_NO_FATAL_FAILURE(ValidateColumnItems(column, ITEMS_COUNT));
+    std::cerr << "Accessing (twice):\t" << timer.GetTotalElapsed().count() << " us"
+              << std::endl;
+
+    Buffer buffer;
+
+    // Save
+    {
+        timer.Reset();
+
+        for (int i = 0; i < LOAD_AND_SAVE_REPEAT_TIMES; ++i)
+        {
+            buffer.clear();
+            BufferOutput bufferOutput(&buffer);
+            CodedOutputStream ostr(&bufferOutput);
+
+            timer.Start();
+            column.Save(&ostr);
+            ostr.Flush();
+            timer.Pause();
+        }
+        const auto elapsed = timer.GetTotalElapsed() / (LOAD_AND_SAVE_REPEAT_TIMES * 1.0);
+
+        std::cerr << "Saving:\t" << elapsed.count() << " us"
+                  << std::endl;
+    }
+
+    std::cerr << "Serialized binary size: " << buffer.size() << std::endl;
+
+    // Load
+    {
+        timer.Reset();
+
+        for (int i = 0; i < LOAD_AND_SAVE_REPEAT_TIMES; ++i)
+        {
+            ArrayInput arrayInput(buffer.data(), buffer.size());
+            CodedInputStream istr(&arrayInput);
+            column.Clear();
+
+            timer.Start();
+            column.Load(&istr, ITEMS_COUNT);
+            timer.Pause();
+        }
+        const auto elapsed = timer.GetTotalElapsed() / (LOAD_AND_SAVE_REPEAT_TIMES * 1.0);
+
+        std::cerr << "Loading:\t" << elapsed.count() << " us"
+                  << std::endl;
+    }
+
+    EXPECT_NO_FATAL_FAILURE(ValidateColumnItems(column, ITEMS_COUNT));
+}
+
+TYPED_TEST_P(ColumnPerformanceTest, InsertAndSelect) {
+    using ColumnType = TypeParam;
+
+    const std::string table_name = "PerformanceTests.ColumnTest";
+    const std::string column_name = "column";
+
+    auto column = InstantiateColumn<ColumnType>();
+    Client client(ClientOptions().SetHost("localhost"));
+    client.Execute("CREATE DATABASE IF NOT EXISTS PerformanceTests");
+    client.Execute("DROP TABLE IF EXISTS PerformanceTests.ColumnTest");
+    client.Execute("CREATE TABLE PerformanceTests.ColumnTest (" + column_name + " " + column.Type()->GetName() + ") ENGINE = Memory");
+
+    const size_t ITEMS_COUNT = 1'000'000;
+
+    std::cerr << "\n===========================================================" << std::endl;
+    std::cerr << "\t" << ITEMS_COUNT << " items of " << column.Type()->GetName()  << std::endl;
+
+    PausableTimer<std::chrono::microseconds> timer;
+
+    timer.Restart();
+    for (size_t i = 0; i < ITEMS_COUNT; ++i)
+    {
+        const auto value = generate(column, i);
+        column.Append(value);
+    }
+
+    EXPECT_EQ(ITEMS_COUNT, column.Size());
+    std::cerr << "Appending:\t" << timer.GetTotalElapsed().count() << " us"
+        << std::endl;
+
+    EXPECT_NO_FATAL_FAILURE(ValidateColumnItems(column, ITEMS_COUNT));
+
+    {
+        Block block;
+        block.AppendColumn(column_name, column.Slice(0, ITEMS_COUNT));
+
+        timer.Restart();
```

Comment:
To minimize the risks of typos, especially in such precise procedure like performance measurement, I think it's better to create new instances of short-lived timer for each sub-case, if you are not going to accumulate measurements. (here and below)

Reply:
done